src/control/server/mgmt_svc_test.go

```diff
@@ -130,3 +141,174 @@ func TestCheckMgmtSvcReplica(t *testing.T) {
 		}
 	}
 }
+
+// mockDrpcClient is a mock of the DomainSocketClient interface
+type mockDrpcClient struct {
+	sync.Mutex
+	ConnectOutputError    error
+	CloseOutputError      error
+	CloseCallCount        int
+	SendMsgInputCall      *drpc.Call
+	SendMsgOutputResponse *drpc.Response
+	SendMsgOutputError    error
+}
+
+func (c *mockDrpcClient) IsConnected() bool {
+	return false
+}
+
+func (c *mockDrpcClient) Connect() error {
+	return c.ConnectOutputError
+}
+
+func (c *mockDrpcClient) Close() error {
+	c.CloseCallCount++
+	return c.CloseOutputError
+}
+
+func (c *mockDrpcClient) SendMsg(call *drpc.Call) (*drpc.Response, error) {
+	c.SendMsgInputCall = call
+	return c.SendMsgOutputResponse, c.SendMsgOutputError
+}
+
+func (c *mockDrpcClient) setSendMsgResponse(status drpc.Status, body []byte) {
+	c.SendMsgOutputResponse = &drpc.Response{
+		Status: status,
+		Body:   body,
+	}
+}
+
+func newTestMgmtSvc(log logging.Logger) *mgmtSvc {
+	r := ioserver.NewRunner(log, ioserver.NewConfig())
+
+	var msCfg mgmtSvcClientCfg
```

Comment:
this logic may be useful for other tests and could be moved out to helper at some point

Reply:
I found this was actually not possible to move out to common. mgmtSvc is an unexported type. Created a server test utils file for helper functions like this that are used within the server.